code/modules/mining/ores_coins.dm

```diff
@@ -380,22 +380,60 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 
 /obj/item/coin/plasma
 	cmineral = "plasma"
+	desc = "You really shouldn't keep this in the same pocket as a lighter."
 	icon_state = "coin_plasma_heads"
 	materials = list(MAT_PLASMA = 400)
 	credits = 80
 
+/obj/item/coin/plasma/bullet_act(obj/item/projectile/P)
+	if(!QDELETED(src) && !P.nodamage && (P.damage_type == BURN))
+		log_and_set_aflame(P.firer, P)
+
+/obj/item/coin/plasma/attackby(obj/item/I, mob/living/user, params)
+	if(is_hot(I))
+		log_and_set_aflame(user, I)
+	return
```

Comment:
```suggestion
/obj/item/coin/plasma/attackby(obj/item/I, mob/living/user, params)
	if(!is_hot(I))
		return
	log_and_set_aflame(user, I)
```
1. This looks cleaner as a guard clause
2. The return at the end of a proc is implicit and not needed

Reply:
it should probably also be return ..()